# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: extensions/permissions/PermissionManager.cpp
# Commit: 202c4f5c642b
# Full Hash: 202c4f5c642b8ec35e849c2144f935342d193fd7
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 3: Screen Wake Lock permission integration. r=webidl,dom-core,smaug,pbz,edgar
#   
#   Depends on D189509
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189510
# ==============================================================================

diff -r 82e03a404c2f -r 202c4f5c642b extensions/permissions/PermissionManager.cpp
--- a/extensions/permissions/PermissionManager.cpp	Tue Dec 05 15:37:05 2023 +0000
+++ b/extensions/permissions/PermissionManager.cpp	Tue Dec 05 15:37:06 2023 +0000
@@ -107,8 +107,8 @@
   // A list of permissions that can have a fallback default permission
   // set under the permissions.default.* pref.
   static const nsLiteralCString kPermissionsWithDefaults[] = {
-      "camera"_ns, "microphone"_ns, "geo"_ns, "desktop-notification"_ns,
-      "shortcuts"_ns};
+      "camera"_ns,    "microphone"_ns,      "geo"_ns, "desktop-notification"_ns,
+      "shortcuts"_ns, "screen-wake-lock"_ns};
 
   if (!aType.IsEmpty()) {
     for (const auto& perm : kPermissionsWithDefaults) {